File: lib/src/main/java/com/yandex/authsdk/internal/Logger.java
package com.yandex.authsdk.internal;

import androidx.annotation.NonNull;
import android.util.Log;

import com.yandex.authsdk.YandexAuthOptions;

public class Logger {

    public static void e(
            @NonNull final YandexAuthOptions options,
            @NonNull final String tag,
            @NonNull final String message,
            @NonNull final Throwable e
    ) {
        if (options.isLoggingEnabled()) {
            Log.e(tag, message, e);
        }
    }

    public static void d(
            @NonNull final YandexAuthOptions options,
            @NonNull final String tag,
            @NonNull final String message
    ) {
        if (options.isLoggingEnabled()) {
            Log.d(tag, message);
        }
    }
}

File: lib/src/main/java/com/yandex/authsdk/internal/Logger.kt
package com.yandex.authsdk.internal

import android.util.Log
import com.yandex.authsdk.YandexAuthOptions

internal object Logger {

    fun e(options: YandexAuthOptions, tag: String, message: String, e: Throwable) {
        if (options.isLoggingEnabled) {
            Log.e(tag, message, e)
        }
    }

    fun d(options: YandexAuthOptions, tag: String, message: String) {
        if (options.isLoggingEnabled) {
            Log.d(tag, message)
        }
    }
}
